tests/rptest/services/redpanda.py

```diff
@@ -1608,6 +1608,88 @@ def get_node_by_id(self, id):
     def stop_node(self, node, **kwargs):
         pass
 
+    def restart_pod(self, pod_name: str, timeout: int = 180):
+        """Restart a pod by name
+
+        Using kubectl delete to gracefully stop the pod,
+        the pod will automatically be restarted by the cluster.
+        Block until pod container is ready.
+        The list of pod names can be found in self.pods property.
+
+        :param pod_name: string of pod name, e.g. 'rp-clo88krkqkrfamptsst0-5'
+        :param timeout: seconds to wait until pod is ready after kubectl delete
+        """
+        self.logger.info(f'restarting pod {pod_name}')
+
+        def pod_container_ready(pod_name: str):
+            # kubectl get pod rp-clo88krkqkrfamptsst0-0 -n=redpanda -o=jsonpath='{.status.containerStatuses[0].ready}'
+            return self._kubectl.cmd([
+                'get', 'pod', pod_name, '-n=redpanda',
+                "-o=jsonpath='{.status.containerStatuses[0].ready}'"
+            ]).decode()
+
+        delete_cmd = ['delete', 'pod', pod_name, '-n=redpanda']
+        self.logger.info(
+            f'deleting pod {pod_name} so the cluster can recreate it')
+        # kubectl delete pod rp-clo88krkqkrfamptsst0-0 -n=redpanda'
```

Comment:
from [docs](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-termination), this sends a `SIGTERM`